LiveSched/src/main/java/dev/coms4156/project/livesched/MyFileDatabase.java

```diff
@@ -0,0 +1,188 @@
+   * @param resourceTypeFilePath the path to the file containing the entries of the database
+   */
+  public MyFileDatabase(int flag, String taskFilePath, String resourceTypeFilePath) {
+    this.taskFilePath = taskFilePath;
+    this.resourceTypeFilePath = resourceTypeFilePath;
+
+    if (flag == 0) {
+      this.allTasks = deSerializeObjectFromFile(taskContentType);
+      this.allResourceTypes = deSerializeObjectFromFile(resourceTypeContentType);
+    } else {
+      this.allTasks = new ArrayList<>();
+      this.allResourceTypes = new ArrayList<>();
+    }
+  }
+
+  /**
+   * Sets allTasks of the database.
+   *
+   * @param tasks the list of all tasks to be added to database
+   */
+  public void setAllTasks(List<Task> tasks) {
+    this.allTasks = tasks == null ? new ArrayList<>() : tasks;
+  }
+
+  /**
+   * Sets allResourceTypes of the database.
+   *
+   * @param resourceTypes the list of all resourceTypes to be added to database
+   */
+  public void setAllResourceTypes(List<ResourceType> resourceTypes) {
+    this.allResourceTypes = resourceTypes == null ? new ArrayList<>() : resourceTypes;
+  }
+
+  /**
+   * Deserializes the object from the file and returns the department mapping.
+   * Throws exception if data in file is invalid
+   *
+   * @return the deserialized department mapping
+   */
+  public final <T> List<T> deSerializeObjectFromFile(int contentType) {
+    String filePath;
+    if (contentType == taskContentType) {
+      filePath = taskFilePath;
+    } else if (contentType == resourceTypeContentType) {
+      filePath = resourceTypeFilePath;
+    } else {
+      throw new IllegalArgumentException("Invalid content type in file.");
+    }
+
+    try (ObjectInputStream in = new ObjectInputStream(new FileInputStream(filePath))) {
+      Object obj = in.readObject();
+      if (obj instanceof List<?> listObj) {
+
+        List<T> finalList = new ArrayList<>();
+
+        for (Object value : listObj) {
+          if (contentType == taskContentType && !(value instanceof Task)) {
+            throw new IllegalArgumentException("Invalid object type in file.");
+          } else if (contentType == resourceTypeContentType && !(value instanceof ResourceType)) {
+            throw new IllegalArgumentException("Invalid object type in file.");
+          }
+          finalList.add((T) value);
+        }
+        return finalList;
+      } else {
+        throw new IllegalArgumentException("Invalid object type in file.");
+      }
+    } catch (IOException | ClassNotFoundException e) {
+      if (LOGGER.isLoggable(Level.SEVERE)) {
+        LOGGER.log(Level.SEVERE, e.getMessage());
+      }
+      return new ArrayList<>();
+    }
+  }
+
+  /**
+   * Saves the contents of the internal data structures to the file. Contents of the file are
+   * overwritten with this operation.
+   *
+   * @param contentType the type of content to be saved to file
+   */
+  public void saveContentsToFile(int contentType) {
```

Comment:
could be nice to indicate options for contentType (i.e. 1 = task, 2 = resourceType) but we can leave that for the future